# Modelos en Machine Learning

Par tener acceso a los diferentes modelos, en Python, utilizaremos la libreria de Scikit Learn.
Esta libreria ya tiene los modelos, preparados para poder entrenar los datos que nosotros tengamos.

In [0]:
import numpy as np 
import pandas as pd 
from sklearn import linear_model

##Importar datos

La libreria de pandas tiene muchas herramientas para importar datos, una de ellas nos permite importar datos desde una fuente externa. En este caso utilizaremos una fuente de datos precargada.

In [5]:
df = pd.read_csv('https://raw.githubusercontent.com/leonpalafox/clasemlMP/master/FundamentosMachineLearning/data/custclv.csv')
df.head()

customer_id  rfm_score Gender  ...       CLV MonthlyValue ActiveMonths
0     10235005        322   Male  ...    7.3780      3.68900            2
1     10238628        535   Male  ...  399.1002    199.55010            2
2     10256519        545   Male  ...  534.6880    534.68800            1
3     10261907        213   Male  ...   17.9180     17.91800            1
4     10269642        554   Male  ...  570.9394    142.73485            4

[5 rows x 11 columns]

## Carga de Modelo
Para crear un modelo de regresión lineal basta con cargar las variables de interés en la función.

In [10]:
df_modelo = df[['rfm_score', 'Age', 'ActiveMonths']]
df_obj = df['CLV']
lr = linear_model.LinearRegression() #lr es el Modelo de regesión lineal
lr.fit(df_modelo, df_obj)
lr.coef_

array([ 0.07896835, -0.62826523, 75.73227722])

Este resultado implica:

\begin{align}
0.0789.rfm\_score -0.6282.Age+75.732277.ActiveMonths = CLV
\end{align}

El lr.fit ejecutó el tercer elemento en el paradigma fundamental de Machine Learning, el cual es la optimización